worker/uniter/relation/resolver_test.go

```diff
@@ -279,6 +279,37 @@ func (s *relationResolverSuite) TestNextOpNothing(c *gc.C) {
 	c.Assert(errors.Cause(err), gc.Equals, resolver.ErrNoOperation)
 }
 
+// This is a regression test for LP1906706
+func (s *relationResolverSuite) TestCreatedRelationsResolverWithPendingHook(c *gc.C) {
+	unitTag := names.NewUnitTag("wordpress/0")
+
+	var numCalls int32
+	unitEntity := params.Entities{Entities: []params.Entity{{Tag: "unit-wordpress-0"}}}
+	unitStateResults := params.UnitStateResults{Results: []params.UnitStateResult{{}}}
+	apiCaller := mockAPICaller(c, &numCalls,
+		uniterAPICall("Refresh", unitEntity, params.UnitRefreshResults{Results: []params.UnitRefreshResult{{Life: life.Alive, Resolved: params.ResolvedNone}}}, nil),
+		uniterAPICall("GetPrincipal", unitEntity, params.StringBoolResults{Results: []params.StringBoolResult{{Result: "", Ok: false}}}, nil),
+		uniterAPICall("RelationsStatus", unitEntity, params.RelationUnitStatusResults{Results: []params.RelationUnitStatusResult{{RelationResults: []params.RelationUnitStatus{}}}}, nil),
+		uniterAPICall("State", unitEntity, unitStateResults, nil),
+	)
+	r := s.newRelationStateTracer(c, apiCaller, unitTag)
+	assertNumCalls(c, &numCalls, 4)
+
+	localState := resolver.LocalState{
+		State: operation.State{
+			Installed: true,
+			Kind:      operation.RunHook,
```

Comment:
This test seems very specific about the precise precondition that would result in NoOp. I'm not sure it encapsulates the idea of "a failed hook should give the right post-condition to result in recording an error".

Reply:
Ideally, this would be one of those odd uniter tests (the ones with their own "DSL" for describing the spec). 

However, given the [way](https://github.com/juju/juju/blob/2.8/worker/uniter/resolver.go#L142-L144) that the uniter changes to an error state  writing a proper test is cumbersome because you have to pretty much set up all resolvers (and mock the API calls that they may potentially make). So instead, I opted to explicitly test the "if hook pending then return ErrNoOperations" logic (which causes the main resolver loop to set the error state) as the lack of this check in the created relations resolver was the reason for the bug.